src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -130,11 +225,12 @@ bool EffectDummyCreature_lazy_peon_awake(Unit* pCaster, uint32 uiSpellId, SpellE
     // always check spellid and effectindex
     if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
     {
+        //TODO some of these checks are now redundant, should be removed by someone who knows how to do so without breaking things
```

Comment:
It is very much possible for uispellId to be wrong, to my understanding, this function;
https://github.com/cmangos/mangos-classic/pull/220/files#diff-e1032aae835a2b320c86b6ff4b63a3a8L128
is executed whenever the peon is hit by **any** spellEffect, what-so-ever, (it's simply a EffectDummyNPC with a poor name; https://github.com/cmangos/mangos-classic/pull/220/files#diff-e1032aae835a2b320c86b6ff4b63a3a8R248) <- as you can see, it's a function defined in the script exclusively. Not used by the core. If you don't use it, reduce it =)

I'd say rather turn it into a void function instead of an int function since the result is not used anywhere in the script anyways, and then perhaps also give it a slightly better name.

to me, I see no reason it couldn't just be;

void EffectDummyCreature_lazy_peon(Unit* pCaster, uint32 uiSpellId, SpellEffectIndex uiEffIndex, Creature* pCreatureTarget, ObjectGuid /*originalCasterGuid*/)
{
// Only handle peon awaken, ignore any other spell effects.
if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
if (npc_lazy_peonAI* pPeonAI = dynamic_cast<npc_lazy_peonAI*>(pCreatureTarget->AI()))
pPeonAI->Awaken(pCaster);
}

but if it for some reason has to be an int function because of sd2 internals then it could be;

void EffectDummyCreature_lazy_peon(Unit* pCaster, uint32 uiSpellId, SpellEffectIndex uiEffIndex, Creature* pCreatureTarget, ObjectGuid /*originalCasterGuid*/)
{
// Only handle peon awaken, ignore any other spell effects.
if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
if (npc_lazy_peonAI* pPeonAI = dynamic_cast<npc_lazy_peonAI*>(pCreatureTarget->AI()))
{
pPeonAI->Awaken(pCaster);
return true;
}

// failed to cast to AI or wrong spell
return false;
}

(sorry for poor formatting)